clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesDataProviderIntegrationTest.java

```diff
@@ -0,0 +1,1117 @@
+    softly.assertThat(serverGroup.getName()).isEqualTo("replicaSet frontend-64545c4c54");
+    softly.assertThat(serverGroup.getInstanceCounts().getUp()).isEqualTo(0);
+    softly.assertThat(serverGroup.getInstanceCounts().getTotal()).isEqualTo(0);
+    softly.assertThat(serverGroup.getLoadBalancers()).containsExactly("service frontend");
+    softly.assertThat(serverGroup.getUid()).isEqualTo("13939207-d970-4e19-8f8d-ffcd353016ff");
+    // When using a deployment, the prior server group is not disabled as labels aren't changed;
+    // instead this server group is scaled down to 0 instances.
+    softly.assertThat(serverGroup.isDisabled()).isFalse();
+    softly.assertThat(serverGroup.getRegion()).isEqualTo("frontend-ns");
+    softly.assertThat(serverGroup.getServerGroupManagers()).hasSize(1);
+    if (!serverGroup.getServerGroupManagers().isEmpty()) {
+      assertFrontEndServerGroupManagerSummary(
+          softly, serverGroup.getServerGroupManagers().iterator().next());
+    }
+    softly
+        .assertThat(serverGroup.getLabels())
+        .containsAllEntriesOf(
+            ImmutableMap.of(
+                "app.kubernetes.io/name", "frontendapp",
+                "app.kubernetes.io/managed-by", "spinnaker"));
+    softly.assertThat(serverGroup.getType()).isEqualTo("kubernetes");
+    softly
+        .assertThat((Collection<String>) serverGroup.getBuildInfo().get("images"))
+        .containsExactly("nginx:1.19.0");
+    softly.assertThat(serverGroup.getZone()).isEqualTo("frontend-ns");
+    softly.assertThat(serverGroup.getCloudProvider()).isEqualTo("kubernetes");
+    softly.assertThat(serverGroup.getInstances()).isEmpty();
+  }
+
+  private void assertFrontendCurrentServerGroup(
+      SoftAssertions softly, KubernetesV2ServerGroup serverGroup) {
+    softly.assertThat(serverGroup.getMoniker().getApp()).isEqualTo("frontendapp");
+    softly.assertThat(serverGroup.getMoniker().getCluster()).isEqualTo("deployment frontend");
+    softly.assertThat(serverGroup.getMoniker().getSequence()).isEqualTo(2);
+    softly.assertThat(serverGroup.getCapacity().getDesired()).isEqualTo(2);
+    softly.assertThat(serverGroup.getAccount()).isEqualTo("my-account");
+    softly.assertThat(serverGroup.getAccountName()).isEqualTo("my-account");
+    softly.assertThat(serverGroup.getKind()).isEqualTo(KubernetesKind.REPLICA_SET);
+    softly.assertThat(serverGroup.getName()).isEqualTo("replicaSet frontend-5c6559f75f");
+    softly.assertThat(serverGroup.getInstanceCounts().getUp()).isEqualTo(2);
+    softly.assertThat(serverGroup.getInstanceCounts().getTotal()).isEqualTo(2);
+    softly.assertThat(serverGroup.getLoadBalancers()).containsExactly("service frontend");
+    softly.assertThat(serverGroup.getUid()).isEqualTo("29630998-bdee-4586-ac64-45223d7ef7d5");
+    // When using a deployment, the prior server group is not disabled as labels aren't changed;
+    // instead this server group is scaled down to 0 instances.
```

Comment:
Can we remove this comment since it only applies to the prior SG?

Reply:
Removed!